exp/api/v1beta1/azuremachinepool_webhook.go

```diff
@@ -249,22 +250,27 @@ func (amp *AzureMachinePool) ValidateOrchestrationMode(c client.Client) func() e
 	return func() error {
 		// Only Flexible orchestration mode requires validation.
```

Comment:
Just fyi. A while back we implemented CustomDefaulter/CustomValidator in controller-runtime.

This should make large parts of the CAPZ webhook util redundant.

Just in case you want to follow-up on that in a separate PR.

Let me know if you need/want more information

Reply:
Thanks for noticing this! I opened #3103.